cmd/relayproxy/config/config.go

```diff
@@ -322,3 +327,45 @@ func locateConfigFile(inputFilePath string) (string, error) {
 	return "", fmt.Errorf(
 		"impossible to find config file in the default locations [%s]", strings.Join(defaultLocations, ","))
 }
+
+// Load the ENV Like:RETRIEVERS_0_HEADERS_AUTHORIZATION
+func loadArrayEnv(s string, v string, configMap map[string]interface{}) error {
+	paths := strings.Split(s, "_")
+	for i, str := range paths {
+		paths[i] = strings.ToLower(str)
+	}
+	prefixKey := paths[0]
+	if configArray, ok := configMap[prefixKey].([]interface{}); ok {
+		index, err := strconv.Atoi(paths[1])
+		if err != nil {
+			return err
+		}
+		configItem := configArray[index].(map[string]interface{})
```

Comment:
Here we can have an error if we try to add an item to the array.

Let's say I want to add a new retriever and not just override it, I will have some env variable like this:
```
RETRIEVERS_1_KIND=github
RETRIEVERS_1_REPOSITORYSLUG=thomaspoignant/go-feature-flag
RETRIEVERS_1_BRANCH=main
RETRIEVERS_1_FILEPATH=testdata/flag-config.yaml
```

And when doing this, the `configArray[index]` will error because the array size is 0 in our case.